### PR TITLE
fix empty cart flashing when registering user

### PR DESCRIPTION
**Pull request by iakhator**
Resolve issue #3036 

This PR fix ```empty cart``` showing up while registering new user,  The reason for the empty cart showing up is that the cart.items is empty at the that window it tries to register the user. What I did was to create a helper method in `checkout.js` called `cartCount` where it checks if the `cart.item` exists and also check if the length is greater than 0.

#### Test
* Place an order.
* Checkout order.
* Next step will show telling you to `register`. 
* Input details and `register`.
* It takes you to the next step.

### CODE REVIEW

**Comment by iakhator** [comment]
cc @rymorgan

**Comment by rymorgan** [comment]
I confirmed it's not flashing anymore, but it's a jarring experience for small action like registering a user to have a spinner happen on the whole page. If a spinner is necessary it should happen on the component level.

**Comment by iakhator** [comment]
@rymorgan, the sad thing about this is that it is written in blaze so  it will be better the spinner is removed in my opinion or what do you suggest I do?

**Comment by rymorgan** [comment]
@zenweasel Thoughts? We should really be working on converting this to react. I don't want to sink a bunch of work into blaze components. This solution looks less broken thane before but I don't think is good UX yet.

**Comment by zenweasel** [comment]
I definitely don't want this bug fix ticket to upscope into rewriting the checkout process in React. If this fixes the original bug let's merge and move on and if we need to open a new issue for the spinner then let's do that but probably we will end up revisiting the entire UI.

**Comment by rymorgan** [comment]
Ok, we can merge this, but I'm not happy with this UX. But it's worth merging because it looks fixes it from looking broken. I'll make an issue to improve this UX upon register.

**Comment by zenweasel** [comment]
If I am on the Completed Order page and I log out I get a cascade of errors:
```
Invariant Violation: Route.render(): A valid React element (or null) must be returned. You may have returned undefined, an array or some other invalid object.
    at invariant (http://localhost:3000/packages/modules.js?hash=7ee907e6d17784321fe48b45f09bcaa0f5d0b366:85884:15)
    at ReactCompositeComponentWrapper._renderValidatedComponent (http://localhost:3000/packages/modules.js?hash=7ee907e6d17784321fe48b45f09bcaa0f5d0b366:108730:143)
    at ReactCompositeComponentWrapper.performInitialMount (http://localhost:3000/packages/modules.js?hash=7ee907e6d17784321fe48b45f09bcaa0f5d0b366:108261:30)
    at ReactCompositeComponentWrapper.mountComponent (http://localhost:3000/packages/modules.js?hash=7ee907e6d17784321fe48b45f09bcaa0f5d0b366:108157:21)
    at Object.mountComponent (http://localhost:3000/packages/modules.js?hash=7ee907e6d17784321fe48b45f09bcaa0f5d0b366:100918:35)
    at ReactCompositeComponentWrapper.performInitialMount (http://localhost:3000/packages/modules.js?hash=7ee907e6d17784321fe48b45f09bcaa0f5d0b366:108270:34)
    at ReactCompositeComponentWrapper.mountComponent (http://localhost:3000/packages/modules.js?hash=7ee907e6d17784321fe48b45f09bcaa0f5d0b366:108157:21)
    at Object.mountComponent (http://localhost:3000/packages/modules.js?hash=7ee907e6d17784321fe48b45f09bcaa0f5d0b366:100918:35)
    at Object.updateChildren (http://localhost:3000/packages/modules.js?hash=7ee907e6d17784321fe48b45f09bcaa0f5d0b366:107721:51)
    at ReactDOMComponent._reconcilerUpdateChildren (http://localhost:3000/packages/modules.js?hash=7ee907e6d17784321fe48b45f09bcaa0f5d0b366:107243:32)
```

**Comment by zenweasel** [comment]
Verified does not happen on `master`

**Comment by iakhator** [comment]
ok, I will see to this

**Comment by iakhator** [comment]
@zenweasel , I have fixed all errors happening on logout and also glitches happening when signing in and signing out.

**Comment by spencern** [comment]
@zenweasel @iakhator 
I'm seeing the following error message when attempting to change the base branch to `release-1.5.10` 

>There are no new commits between base branch 'release-1.5.10' and head branch 'itua-fix-issue-3036'

is it possible that this PR is a duplicate of something else?

**Comment by zenweasel** [comment]
@spencern Even if the code was duplicated, wouldn't the commits still be unique?

File: imports/plugins/core/checkout/client/templates/checkout/checkout.html [hunk 1]
```diff
@@ -25,7 +25,15 @@
       </div>
     </div>
   {{else}}
-    {{> emptyCheckoutCart}}
+    {{#if currentUser}}
```
Comment (spencern) [inline imports/plugins/core/checkout/client/templates/checkout/checkout.html]: image

While this does seem to work to remove the flash of "your cart is empty" I'm seeing a side effect of causing an "endless" spinner to be rendered for any logged in user who happens to be on the `cart/checkout` route with an empty cart.

Comment (iakhator) [reply imports/plugins/core/checkout/client/templates/checkout/checkout.html]: @spencern, @zenweasel 
The reason for this is happens is the way I implemented the checks. The user is shown with an infinite spinner which was a wrong implementation regarding the fact that every logged in user should be able to see the `empty cart page` if there are not items in the cart. 

#### Fix
- I created a helper method which checks if the items in the cart is available and if it has a value.

#### Test
- Login
- go to `cart/checkout` route.
- `empty cart error` should show now
- Checks works fine now. It register the user without flashing an empty cart

File: imports/plugins/included/product-variant/containers/gridItemControlsContainer.js [hunk 2]
```diff
@@ -44,10 +44,20 @@ const wrapComponent = (Comp) => (
     // check whether all required fields have been submitted before publishing
     checkValidation = () => {
       // this returns an array with a single object
-      const variants = ReactionProduct.getVariants(this.props.product._id).map((variant) => this.validation.validate(variant));
-      this.setState({
-        validProduct: Object.assign({}, this.props.product, { __isValid: variants[0].isValid })
-      });
+      const variants = ReactionProduct.getVariants(this.props.product._id);
```
Comment (zenweasel) [inline imports/plugins/included/product-variant/containers/gridItemControlsContainer.js]: What's happening here?

File: package.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "reaction",
   "description": "Reaction is a modern reactive, real-time event driven ecommerce platform.",
-  "version": "1.5.10",
+  "version": "1.5.11",
```
Comment (zenweasel) [inline package.json]: What?

**Pull request closed without merging**